### DISCUSSION TITLE
FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - KeyError: 'eth_usd_price_feed'

### DISCUSSION THREAD

**Discussion by hiteshgupta2507** [smartcontractkit/full-blockchain-solidity-course-py#1599] [Q&A]
Hello,

I am getting an error when I just use
> brownie test 
`FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - KeyError: 'eth_usd_price_feed'`

but the test runs file if I use ganache chain
> brownie test --network ganache-fundme

The code is here:
https://github.com/hiteshgupta2507/Brownie_Fund_Me/tree/master/PatrickCollins/demos/brownie_fund_me

Thanks!

**Comment by cromewar** [comment]
Hello @hiteshgupta2507 
The page you send over cannot be accessed, however the most likely scenario is you have a typo on `eth_usd_price_feed` so check it out on your `brownie-config`, also remember you have to add the price feed for each network you want to deploy to.

**Reply by hiteshgupta2507** [reply]
Hi @cromewar 

Thanks for your reply. Could you try now. It's public. I can deploy the contract easily to rinkeby network so that makes me wonder there shouldn't be any typo. The brownie-config.yaml file matches exactly with Patrick's git repo.

I can run the test by doing this 
>brownie test --network ganache-fundme

I get the error only when I use
> brownie test
FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - KeyError: 'eth_usd_price_feed'

**Reply by cromewar** [reply]
So as you have said you can deploy correctly to `rinkeby` the reason is you have `eth_usd_price_feed` for that network, but you are running the test on `ganache-fundme` and as you can see, you don't have the price feed for that network.

![image](https://user-images.githubusercontent.com/20199180/180795644-eba71d3c-62c0-4ccd-8343-98cde1dbc7a7.png)


as ganache is for local testing, you should just duplicate the same key for rinkeby on that network and this should work.
Cheers.

**Reply by hiteshgupta2507** [reply]
> brownie test --network ganache-fundme 
the above line works completely fine


I get error with 
> brownie test

**Reply by cromewar** [reply]
Could you please provide more details of the error please?

**Reply by hiteshgupta2507** [reply]
This is the error I get
```

    def deploy_fund_me():
        account = get_account()
        print(network.show_active())
        # pass the priceFeed address to the FundMe contract
    
        # if we are on a persistent network like rinkeby, use the associated address "0x8A753747A1Fa494EC906cE90E9f37563A8AF630e"
        # otherwise, deploy mocks
        if network.show_active() not in LOCAL_BLOCKCHAIN_DEVELOPMENT:
>           price_feed_address = config["networks"][network.show_active()][
                "eth_usd_price_feed"
            ]
E           KeyError: 'eth_usd_price_feed'

scripts/deploy.py:17: KeyError
---------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------
development
============================================================== short test summary info ===============================================================
FAILED tests/test_fund_me.py::test_can_fund_and_withdraw - KeyError: 'eth_usd_price_feed'`
```

**Reply by cromewar** [reply]
Did you duplicate the eth_usd_price_feed on the other network? development and ganache-fundme?

**Reply by hiteshgupta2507** [reply]
No. I didn't. The eth_usd_price_feed only exists in rinkeby network. 
That's why I was surprise that test on ganache-fundme doesn't throw an error 

but >brownie test   does.
